src/main/java/com/mraof/minestuck/blockentity/machine/AnthvilBlockEntity.java

```diff
@@ -0,0 +1,264 @@
+	 */
+	private void processContents()
+	{
+		if(canBeRefueled() && itemHandler.getStackInSlot(1).is(ExtraForgeTags.Items.URANIUM_CHUNKS))
+		{
+			addFuel((short) FUEL_INCREASE);
+			itemHandler.extractItem(1, 1, false);
+		}
+	}
+	
+	public static void attemptMend(AnthvilBlockEntity anthvil, ServerPlayer player)
+	{
+		Level level = anthvil.level;
+		ItemStack slotStack = anthvil.itemHandler.getStackInSlot(0);
+		GristCache playerCache = GristCache.get(player);
+		
+		if(level == null || !isMendableItem(slotStack) || !hasEnoughFuel(anthvil))
+			return;
+		
+		GristSet pickedGrist = mendingGrist(playerCache, level, slotStack);
+		
+		if(!pickedGrist.isEmpty())
+		{
+			if(slotStack.hasCraftingRemainingItem())
+			{
+				anthvil.itemHandler.setStackInSlot(0, slotStack.getCraftingRemainingItem());
+			} else
+			{
+				//amount of repair changes with the value of the grist, 10 at base
+				int repairAmount = (int) (10 * pickedGrist.asAmounts().get(0).getValue());
+				slotStack.setDamageValue(slotStack.getDamageValue() - repairAmount);
+				
+				if(!player.isCreative() && playerCache.tryTake(pickedGrist, GristHelper.EnumSource.CLIENT))
+				{
+					anthvil.fuel -= MEND_FUEL_COST;
+				}
+			}
+		}
+	}
+	
+	private static boolean isMendableItem(ItemStack slotStack)
+	{
+		return !slotStack.isEmpty() && slotStack.isRepairable() && slotStack.isDamageableItem() && slotStack.isDamaged();
+	}
+	
+	private static boolean hasEnoughFuel(AnthvilBlockEntity anthvil)
+	{
+		return anthvil.fuel >= MEND_FUEL_COST;
+	}
+	
+	/**
+	 * Checks a mend-able item's grist type and returns the grist if its valid and the player can afford to spend the amount
+	 */
+	private static ImmutableGristSet mendingGrist(GristCache playerCache, Level level, ItemStack slotStack)
+	{
+		GristSet fullSet = GristCostRecipe.findCostForItem(slotStack, null, false, level);
+		
+		if(fullSet != null && !fullSet.isEmpty())
+		{
+			GristAmount pickedGrist = getUsedGrist(fullSet);
+			
+			if(playerCache.canAfford(pickedGrist))
+				return pickedGrist;
+		}
+		
+		return GristSet.EMPTY;
+	}
+	
+	/**
+	 * Takes the GristSet of the item stored in the mending slot of the anthvil, finds the most used grist type, then returns a GristSet composed of one value of said grist type.
+	 */
+	public static GristAmount getUsedGrist(GristSet fullSet)
+	{
+		List<GristAmount> gristAmounts = fullSet.asAmounts();
+		GristAmount defaultGrist = new GristAmount(GristTypes.BUILD.get(), 1);
+		
+		if(gristAmounts.size() > 1) //if theres more than one, establishes which GristAmount has the highest impact and uses that
+		{
+			GristAmount pickedGrist = fullSet.asAmounts().stream().max(Comparator.comparingDouble(AnthvilBlockEntity::getModifiedGristValue)).orElse(defaultGrist);
+			return new GristAmount(pickedGrist.type(), 1);
+		} else if(gristAmounts.get(0).type() != GristTypes.BUILD.get())
+			return new GristAmount(gristAmounts.get(0).type(), 1);
+		else
+			return defaultGrist;
```

Comment:
```suggestion
		GristType pickedGrist = fullSet.asAmounts().stream().max(Comparator.comparingDouble(AnthvilBlockEntity::getModifiedGristValue)).map(GristAmount::type).orElse(GristTypes.BUILD.get());
		return new GristAmount(pickedGrist, 1);
```
The stream will work equally well independent on the number of grist amounts, so this stream could replace the other if-cases too. Added another step in this suggestion too.